catboost/libs/documents_importance/docs_importance_helpers.cpp

```diff
@@ -21,16 +23,29 @@ TVector<TVector<double>> TDocumentImportancesEvaluator::GetDocumentImportances(c
     localExecutor.ExecRange([&] (int treeId) {
         leafIndices[treeId] = BuildIndicesForBinTree(Model, binarizedFeatures, treeId);
     }, NPar::TLocalExecutor::TExecRangeParams(0, TreeCount), NPar::TLocalExecutor::WAIT_COMPLETE);
-
     UpdateFinalFirstDerivatives(leafIndices, pool);
-    TVector<TVector<double>> documentImportances(DocCount, TVector<double>(pool.Docs.GetDocCount()));
 
-    localExecutor.ExecRange([&] (int docId) {
-        // The derivative of leaf values with respect to train doc weight.
-        TVector<TVector<TVector<double>>> leafDerivatives(TreeCount, TVector<TVector<double>>(LeavesEstimationIterations)); // [treeCount][LeavesEstimationIterationsCount][leafCount]
-        UpdateLeavesDerivatives(docId, &leafDerivatives);
-        GetDocumentImportancesForOneTrainDoc(leafDerivatives, leafIndices, &documentImportances[docId]);
-    }, NPar::TLocalExecutor::TExecRangeParams(0, DocCount), NPar::TLocalExecutor::WAIT_COMPLETE);
+    TVector<TVector<double>> documentImportances(DocCount, TVector<double>(pool.Docs.GetDocCount()));
+    const ui32 docDivider = 1000;
+    size_t docBlockSize = DocCount < docDivider ? CB_THREAD_LIMIT : CB_THREAD_LIMIT * (CeilDiv(DocCount, docDivider));
```

Comment:
Even simpler: `const size_t docBlockSize = 1000;`, and remove `docDivider`.

Reply:
I have not got this moment. How does it work in case when we have docs amount smaller then 1000 ?

Reply:
If DocCount < docBlockSize, we have just one iteration in the for loop.